### PR TITLE
Address CodeQL vulnerabilities

### PR DESCRIPTION
**Pull request by tpitts-CFI**
Task: Address CodeQL vulnerabilities (3a589da0-3e4e-11ec-976e-2df811d36180)

* CodeQL will show no security vulnerabilities

### MERGE ORDER
#39 
then #34

### CODE REVIEW

File: kohese-portal/server/boot/routes.ts [hunk 1]
```diff
@@ -54,6 +55,15 @@ module.exports = function (app) {
       res.sendFile(path.resolve(clientBundlePath, 'index.html'));
     });
 
+    // set up rate limiter: maximum of five requests per minute
+    var limiter = new RateLimit({
+      windowMs: 1*60*1000, // 1 minute
+      max: 5
```
Comment (cfi-dphillips) [inline kohese-portal/server/boot/routes.ts:61]: Why just 5/min?  If we are doing a repo sync at startup, you could get 5 requests in the first second.  Need to choose a number that is reasonable, but not too high/low...

Comment (tpitts-CFI) [reply kohese-portal/server/boot/routes.ts:57]: I left it at the default. I also tested a repo sync at start up and it didn't have any trouble with it.

Express Rate Limit:
"Basic rate-limiting middleware for Express. Use to limit repeated requests to public APIs and/or endpoints such as password reset."

Comment (cfi-dphillips) [reply kohese-portal/server/boot/routes.ts:57]: Try doing it after a commit has been performed on a different computer. There should be multiple requests required to get back in sync from a computer that was not attached when the commit was performed.  Note the number of requests will increase based on the depth of the change in the tree. So if it is 6 or 7 levels deep, there will be 1-2 requests per level.

Comment (cfi-dphillips) [reply kohese-portal/server/boot/routes.ts:57]: Imagine if you are not online for a week and there are 100 different trees that are updated.  You would have to wait 20 minutes to get in sync...

Comment (tpitts-CFI) [reply kohese-portal/server/boot/routes.ts]: express-rate-limit Only effects HTML requests not socket requests

**Review by cfi-dphillips — APPROVED** [review]

**Pull request merged**